contracts/UniswapV3Staker.sol

```diff
@@ -136,18 +136,9 @@ contract UniswapV3Staker is IUniswapV3Staker, Multicall {
         uint256 endTime,
         address refundee,
         uint256 reward,
-        address token0,
-        address token1,
+        IUniswapV3Pool pool,
         uint24 fee
```

Comment:
I don't think we need `uint24 fee` anymore as a param to the function now that `IUniswapV3Pool pool` is passed in. Could just do:

```
    function createIncentiveWithMaxRange(
        IERC20Minimal rewardToken,
        IUniswapV3Pool pool,
        uint256 startTime,
        uint256 endTime,
        address refundee,
        uint256 reward
    ) external override {
        int24 tickSpacing = factory.feeAmountTickSpacing(pool.fee());
        require(tickSpacing > 0, 'UniswapV3Staker::createIncentiveWithMaxRange: !fee');

        // full range max/min ticks for pool
        int24 maxTick = TickMath.MAX_TICK - (TickMath.MAX_TICK % tickSpacing);
        int24 minTick = -maxTick;

        // use max range for min width
        int24 minWidth = maxTick - minTick; // which is just 2 * maxTick

        IncentiveKey memory key = IncentiveKey(rewardToken, pool, startTime, endTime, minWidth, refundee);

        createIncentive(key, reward);
    }
``` 

Realized we may not need to query the `factory` for the tick spacing as a sanity check on the `pool` (and therefore `pool.tickSpacing()`) as L360 does this, but we may want to keep getting tickSpacing from factory as above just in case.

Reply:
Changed the order of function params to be in line with `IncentiveKey` as well (above), but up to you if you want that.

Reply:
Updated function params order in commit `[ab82b6f411f1b6aa236903ff5aff8a0a9faada22](https://github.com/overlay-market/v3-staker/pull/12/commits/ab82b6f411f1b6aa236903ff5aff8a0a9faada22)`

Reply:
Removed `fee` param and updated to pull tick spacing from factory contract: [59522b32f4edf77249965c5968cc49353247d5c8](https://github.com/overlay-market/v3-staker/pull/12/commits/59522b32f4edf77249965c5968cc49353247d5c8)